snapcraft/projects.py

```diff
@@ -408,6 +408,24 @@ def unmarshal(cls, data: Dict[str, Any]) -> "Project":
 
         return project
 
+    def _get_content_plugs(self) -> List[ContentPlug]:
+        """Get list of content plugs."""
+        if self.plugs is not None:
+            return [
+                plug for plug in self.plugs.values() if isinstance(plug, ContentPlug)
+            ]
+        return []
+
+    def get_content_snaps(self) -> Optional[List[str]]:
+        """Get list of snaps from ContentPlug `default-provider` fields."""
+        content_snaps = [
+            x.default_provider
+            for x in self._get_content_plugs()
+            if x.default_provider is not None
+        ]
+
+        return content_snaps if content_snaps else None
```

Comment:
if this essentially returned content_snaps the cascading logic would be simpler, this should do for now